lib/dml/marketplace/bounty_state_machine.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Dml.Marketplace.BountyStateMachine do
+  alias Dml.Marketplace.Bounty
+  alias Dml.Repo
+
+  use Machinery,
+    states: ["pending", "open", "closed", "finished"],
+    transitions: %{
+      "pending" => "open",
+      "open" => "closed",
+      "*" => "finished"
+    }
+
+  def persist(struct, next_state) do
+    {:ok, bounty} =
+      struct
+      |> Bounty.update_state_changeset(%{state: next_state})
+      |> Repo.update()
+
+    bounty
+  end
+end
```

Comment:
@chikeichan this is - probably - the most important piece, this is the state machine for the `Bounty` object.

There was no mockups/layouts for the bounty creation, so this is my suggestion of possible states & transitions.